scanomatic/data_processing/pheno/save.py

```diff
@@ -127,7 +129,10 @@ def save_state(
         or not os.path.isfile(p)
         or _do_ask_overwrite(p)
     ):
-        np.save(p, settings.serialized())
+        with open(p, 'w') as fp:
+            json.dump(json.loads(
+                jsonizer.dumps(settings.asdict()),
+            ), fp)
```

Comment:
I'm not proud of this, but not sure in what way it would be reasonable to generalize the jsonizer... Suggestions welcome!

Reply:
`jsonizer` has a `dump` too. also I don't know if we need an extra asdict on settings, cant we use asdict imported from dataclasses directly? Finally I think it would be more in line with the intent of the jsonizer if convertion from settings to dict happened inside jsonizer.dumps just like the enums and models are treated.

Reply:
I tried using the jsonizerd.dump, but how the jsonizer needs to be extended/generalized to accommodate this type of object seemed to be non-trivial. I agree it should be done, but I would like to discuss how, before I set off in the wrong direction. It only expects `Models` at the moment.

Reply:
dump(s) should perhaps rightly accept `Any`